### PR TITLE
Add FAQ on how to add a message attachment via the API

### PR DESCRIPTION
**Pull request by jasonblais**
Not part of this PR, but we should at some point move the dev-focused integration documentation from https://docs.mattermost.com/guides/integration.html to developers.mattermost.com

### CODE REVIEW

File: source/developer/message-attachments.rst [hunk 1]
```diff
@@ -124,10 +124,33 @@ And here is how it renders in Mattermost:
...
+---------------------------
+
+Can I send a message attachment via the API?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes, you can use the `create post RESTful API <https://api.mattermost.com/#tag/posts%2Fpaths%2F~1posts%2Fpost>`_.
+
+You need to add an "attachment" key to the post's props JSON field. The value is an array of message attachments you want attached to the post. See below for an example curl command.
+
+.. code-block:: bash
```
Comment (crspeller) [inline source/developer/message-attachments.rst]: Why are there two similar examples?

Comment (jasonblais) [reply source/developer/message-attachments.rst]: One with bash, another with an HTTP request, like in our incoming webhook dev doc example: https://developers.mattermost.com/integrate/incoming-webhooks/#basic-usage

Comment (crspeller) [reply source/developer/message-attachments.rst]: The second example is not an HTTP request the proper example for the curl command is:
```http
POST /posts HTTP/1.1
Host: {your-mattermost-site}
User-Agent: curl/7.63.0
Accept: */*
Content-Type: application/json
Content-Length: 192

{"channel_id":"qmd5oqtwoibz8cuzxzg5ekshgr", "message":"Test message #testing", "props":{"attachments": [{"pretext": "This is the attachment pretext.","text": "This is the attachment text."}]}}
```

Comment (jasonblais) [reply source/developer/message-attachments.rst]: Thanks Chris! Updated doc

Comment (crspeller) [reply source/developer/message-attachments.rst]: I was unclear, sorry. You want to keep the old curl example and replace the HTTP request example with the above.

Comment (jasonblais) [reply source/developer/message-attachments.rst]: That makes more sense--like so?

Comment (crspeller) [reply source/developer/message-attachments.rst]: Yes!

**Pull request merged**